notebooks/simple_xgboost_example.ipynb

```diff
@@ -0,0 +1,714 @@
+    "  },\n",
+    "  {\n",
+    "    key: \"threshold\"\n",
+    "    value: { string_value: \"0.5\" }\n",
+    "  },\n",
+    "  {\n",
+    "    key: \"algo\"\n",
+    "    value: { string_value: \"ALGO_AUTO\" }\n",
+    "  },\n",
+    "  {\n",
+    "    key: \"storage_type\"\n",
+    "    value: { string_value: \"AUTO\" }\n",
+    "  },\n",
+    "  {\n",
+    "    key: \"blocks_per_sm\"\n",
+    "    value: { string_value: \"0\" }\n",
+    "  }\n",
+    "]\n",
+    "\n",
+    "dynamic_batching {\n",
+    "  preferred_batch_size: [1024, 8192]\n",
+    "  max_queue_delay_microseconds: 100\n",
+    "}\n",
+    "EOL"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, the model repository should look like this:\n",
+    "\n",
+    "```\n",
+    "model_repository/\n",
+    "`-- fil\n",
+    "    |-- 1\n",
+    "    |   `-- xgboost.model\n",
+    "    `-- config.pbtxt\n",
+    "```\n",
+    "\n",
+    "**3. Deploy the model in Triton Inference Server**\n",
+    "\n",
+    "Finally, we can deploy the xgboost model in Triton Inference Server using the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run the Triton Inference Server in a Subprocess from Jupyter notebook\n",
+    "\n",
+    "# The os.setsid() is passed in the argument preexec_fn so\n",
+    "# it's run after the fork() and before  exec() to run the shell.\n",
+    "pro = subprocess.Popen([\"tritonserver --model-repository=/data/model_repository\"], stdout=subprocess.PIPE, \n",
+    "                       shell=True, preexec_fn=os.setsid) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above command should load the model and print the log `successfully loaded 'fil' version 1`. Triton server listens on the following endpoints:\n",
+    "\n",
+    "```\n",
+    "Port 8000    -> HTTP Service\n",
+    "Port 8001    -> GRPC Service\n",
+    "Port 8002    -> Metrics\n",
+    "```\n",
+    "\n",
+    "We can test the status of the server connection by running the curl command: `curl -v <IP of machine>:8000/v2/health/ready` which should return `HTTP/1.1 200 OK`\n",
+    "\n",
+    "**NOTE:-** In our case the IP of machine on which Triton Server and this notebook are currently running is `localhost`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!curl -v localhost:8000/v2/health/ready"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Stopping Triton Server before proceeding further\n",
+    "os.killpg(os.getpgid(pro.pid), signal.SIGTERM)  # Send the signal to all the process groups"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Concurrent Model Execution\n",
```

Comment:
We should probably signpost this a bit better as well. Maybe it would be worth some sort of outline before we launch into everything explaining that we will be bringing the server up and down several times with different configuration options to demonstrate different features of the server? At the very least, while the server is up for the first time, we should fire off an inference request to demonstrate that it is a functional deployment.